File: frontend/src/app/documents/models/document.ts
import { Brand } from "@core/models/brand";
import { IDocumentType } from "./document-type";
import { IDocumentKeyword } from "./document-keyword";
import { IDocumentLink } from "./document-link";

export type DocumentId = Brand<string,'document-id'>;

export type DocumentName = Brand<string, 'document-name'>;

export type DocumentNumber = Brand<string, 'document-number'>;

export type DocumentDate = Brand<string, 'document-date'>;

export interface IDocument {
    id: DocumentId;
    name: DocumentName;
    number: DocumentNumber;
    documentType: IDocumentType;
    releaseDate: DocumentDate;
    takeEffectDate: DocumentDate;
    keywords: IDocumentKeyword [];
    links ?: IDocumentLink [];
}

File: Backend/FindDox.Models/Api/Document.cs
namespace FindDox.Models.Api;

public class Document
{
	public Guid Id { get; set; }

	public string Name { get; set; }
	public string Number { get; set; }

	public DateTimeOffset ReleaseDate { get; set; }
	public DateTimeOffset TakeEffectDate { get; set; }


	public DocumentType DocumentType { get; set; }
	public IReadOnlyList<Keyword> Keywords { get; set; }
	public IReadOnlyList<Link> Links { get; set; }
}
